server/src/main/java/org/apache/druid/metadata/IndexerSQLMetadataStorageCoordinator.java

```diff
@@ -660,6 +662,18 @@ private Map<SegmentIdWithShardSpec, SegmentIdWithShardSpec> upgradePendingSegmen
           : overlappingPendingSegments.entrySet()) {
         final SegmentIdWithShardSpec pendingSegmentId = overlappingPendingSegment.getKey();
         final String pendingSegmentSequence = overlappingPendingSegment.getValue();
+
+        boolean considerSequence = false;
```

Comment:
Based on the offline discussion, I think we had decided to include the sequence name in the SELECT query itself, rather than filtering in-memory here.

Reply:
Done